tests/genre.test.js

```diff
@@ -0,0 +1,152 @@
+const { expect } = require('chai');
+const request = require('supertest');
+const { Genre } = require('../src/models');
+const app = require('../src/app');
+
+describe('/genres', () => {
+    before(async () => Genre.sequelize.sync());
+
+    beforeEach(async () => {
+        await Genre.destroy({ where: {} });
+    });
+
+    describe('with no recrods in the database', () => {
+        describe('POST /genres', () => {
+            it('creates a new genre in the database', async () => {
+                const response = await request(app).post('/genres').send({
+                    genre: 'Romance'
+                });
+                const newGenreRecord = await Genre.findByPk(response.body.id, {
+                    raw: true,
+                });
+
+                expect(response.status).to.equal(201);
+                expect(response.body.genre).to.equal('Romance');
+                expect(newGenreRecord.genre).to.equal('Romance');
+            });
+
+            it('errors if genre is an empty string', async () => {
+                const response = await request(app).post('/genres').send({
+                    genre: '',
+                });
+                const newGenreRecord = await Genre.findByPk(response.body.id);
+
+                expect(response.status).to.equal(400);
+                expect(response.body).to.haveOwnProperty('errors');
+                expect(newGenreRecord).to.equal(null);
+            });
+
+            it('errors if genre is null', async () => {
+                const response = await request(app).post('/genres').send({
+                    genre: null,
+                });
+                const newGenreRecord = await Genre.findByPk(response.body.id);
+
+                expect(response.status).to.equal(400);
+                expect(response.body).to.haveOwnProperty('errors');
+                expect(newGenreRecord).to.equal(null);
+            });
+        });
+    });
+
+    describe('with records in the database', () => {
+        let genres;
+
+        beforeEach(async () => {
+            await Genre.destroy({ where: {} });
+
+            genres = await Promise.all([
+                Genre.create({
+                    genre: 'Science-fiction',
+                }),
+                Genre.create({
+                    genre: 'Romance',
+                }),
+            ]);
+        });
+
+        describe('POST /genres', () => {
+            it('should return 400 error if the genre is already in the database', async () => {
+                const genreData = genres[0].genre;
+                const response = await request(app).post('/genres').send(genreData);
```

Comment:
Dont need to create the extra variable here:
```suggestion
                const response = await request(app).post('/genres').send(genres[0].genre);
```